### PR TITLE
CPM-859: Add FamiliesSelector component

### PR DESCRIPTION
**Pull request by pierallard**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/front/src/feature/components/FamiliesSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, {FC} from 'react';
```
Comment (Beninho) [inline components/identifier-generator/front/src/feature/components/FamiliesSelector.tsx:1]:
```suggestion
import React, {FC, useState, useEffect} from 'react';
```
Why not import useState & useEffect methods, like in _**useGetFamilies.ts**_ file ?

File: components/identifier-generator/front/src/feature/components/FamiliesSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    if (error instanceof Unauthorized) {
+      return <Helper level={'error'}>{translate('pim_error.unauthorized_list_families')}</Helper>;
+    }
+    return <Helper level={'error'}>{translate('pim_error.general')}</Helper>;
+  }
+
+  return (
+    <MultiSelectInput
+      emptyResultLabel={translate('pim_common.no_result')}
+      placeholder="Please select at least one family"
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/components/FamiliesSelector.tsx:52]: missing translation

File: components/identifier-generator/front/src/feature/hooks/useGetFamilies.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  const [hasSearchChanged, setHasSearchChanged] = React.useState<boolean>(false);
+
+  const {data, isLoading} = useGetFamilies({page, search});
+
+  useEffect(() => {
+    if (typeof data !== 'undefined') {
+      if (data.length < DEFAULT_LIMIT_PAGINATION) {
+        setHasNextPage(false);
+      }
+      setFamilies(oldFamily => (hasSearchChanged ? data : [...(oldFamily || []), ...data]));
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/hooks/useGetFamilies.ts:54]: formerFamilies

File: components/identifier-generator/front/src/feature/hooks/useGetFamilies.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    },
+  });
+};
+
+const usePaginatedFamilies = () => {
+  const [page, setPage] = React.useState<number>(1);
+  const [hasNextPage, setHasNextPage] = React.useState<boolean>(true);
+  const [families, setFamilies] = React.useState<Family[] | undefined>();
+  const [search, setSearch] = React.useState<string>('');
+  const [hasSearchChanged, setHasSearchChanged] = React.useState<boolean>(false);
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/hooks/useGetFamilies.ts:45]: setHasSearchChanged is only called with true

**Review by mmetayer — APPROVED** [review]
mob reveiwed :heavy_check_mark:

**Review by Beninho — APPROVED** [review]

**Review by Balamung — APPROVED** [review]

**Pull request merged**